src/data/nat/basic.lean

```diff
@@ -75,9 +75,11 @@ instance : linear_ordered_comm_monoid_with_zero ℕ :=
   ..nat.linear_ordered_semiring,
   ..(infer_instance : comm_monoid_with_zero ℕ)}
 
-instance : ordered_comm_semiring ℕ := { .. nat.comm_semiring, .. nat.linear_ordered_semiring }
+instance : ordered_cancel_comm_semiring ℕ := { ..nat.comm_semiring, ..nat.linear_ordered_semiring }
 
 /-! Extra instances to short-circuit type class resolution -/
+instance : ordered_semiring ℕ := ordered_cancel_semiring.to_ordered_semiring'
+instance : ordered_comm_semiring ℕ := ordered_cancel_comm_semiring.to_ordered_comm_semiring'
 instance : add_comm_monoid nat    := by apply_instance
```

Comment:
Why break the pattern here? (And put `ordered_semiring` on top?)
```suggestion
instance : ordered_semiring ℕ     := by apply_instance
instance : ordered_comm_semiring ℕ := by apply_instance
instance : add_comm_monoid nat    := by apply_instance
```

Reply:
Oops! This was computability debugging.

Reply:
Sorry, this is not computability debugging at all. It's necessary to avoid assuming `classical.choice` needlessly in `nat` lemmas. `infer_instance` would come up with `ordered_cancel_semiring.to_ordered_semiring` (classical) instead of `ordered_cancel_semiring.to_ordered_semiring'` (intuitionistic).